### ISSUE TITLE
Expose ObjectMetadata and ACL in uploadStreamAsFile

### ISSUE THREAD

**Issue by Voltir** [MfgLabs/commons-aws#13]
Locally, I modified the definition of upload to this, which is not correct in general but solves my immediate issue:

``` scala
  def uploadStreamAsFile(bucket: String, key: String, ctype: String, chunkUploadConcurrency: Int = 1): Flow[ByteString, CompleteMultipartUploadResult, Unit] = {
    import scala.collection.JavaConversions._

    val uploadChunkSize = 8 * 1024 * 1024 // recommended by AWS

    def initiateUpload(bucket: String, key: String, ctype: String): Future[String] = {
      val meta = new ObjectMetadata()
      meta.setContentType(ctype)
      val req =
        new InitiateMultipartUploadRequest(bucket, key, meta)
          .withCannedACL(CannedAccessControlList.PublicRead)
      client.initiateMultipartUpload(req).map(_.getUploadId)
    }
```

Basically, I wanted to set the ctype of the upload, which was done by creating a new ObjectMetadata object, as well as change the default ACL used when uploading the file. Ideally, both of these things would be exposed in the signature of uploadStreamAsFile although im not quite sure what would be the best way - for the object meta data it should be pretty easy, something the following should work:

``` scala
def uploadStreamAsFile(bucket: String, key: String, ctype: String, chunkUploadConcurrency: Int = 1, metadata = new ObjectMetadata()): Flow[ByteString, CompleteMultipartUploadResult, Unit] = {
```

Exposing the ACL though seems more complicated. Perhaps the most general thing to do would be instead of doing the above, have a variant of uploadStreamAsFile simply take a "InitiateMultipartUploadRequest" object as it signature:

``` scala
def uploadStreamAsFile(upload: InitiateMultipartUploadRequest, chunkUploadConcurrency: Int = 1): Flow[ByteString, CompleteMultipartUploadResult, Unit] = {
```

**Comment by Timshel** [comment]
Hey, sorry for the delay. I just added to `dev` a version with your second suggestion :
```scala
def uploadStreamAsFile(upload: InitiateMultipartUploadRequest, chunkUploadConcurrency: Int): Flow[ByteString, CompleteMultipartUploadResult, Unit]
```

**Comment by Timshel** [comment]
I just released `0.12.2` which include this modification.
